LTHPasscodeViewController/LTHPasscodeViewController.m

```diff
@@ -30,32 +28,25 @@
 [[NSBundle bundleWithPath:[[NSBundle bundleForClass:[LTHPasscodeViewController class]] pathForResource:@"LTHPasscodeViewController" ofType:@"bundle"]] localizedStringForKey:(key) value:@"" table:_localizationTableName]
 #endif
 
-// MARK: Please read
-/*
- Using windows[0] instead of keyWindow due to an issue with UIAlertViews / UIActionSheets - displaying the lockscreen when an alertView / actionSheet is visible, or displaying one after the lockscreen is visible results in a few cases:
- * the lockscreen and the keyboard appear on top the av/as, but
-   * the dimming of the av/as appears on top the lockscreen;
-   * if the app is closed and reopened, the order becomes av/as - lockscreen - dimming - keyboard.
- * the lockscreen always appears behind the av/as, while the keyboard
-   * doesn't appear until the av/as is dismissed;
-   * appears on top on the av/as - if the app is closed and reopened with the av/as visible.
- * the lockscreen appears above the av/as, while the keyboard appears below, so there's no way to enter the passcode.
- 
- The current implementation shows the lockscreen behind the av/as.
- 
- Relevant links:
- * https://github.com/rolandleth/LTHPasscodeViewController/issues/16
- * https://github.com/rolandleth/LTHPasscodeViewController/issues/164 (the description found above)
- * https://stackoverflow.com/questions/19816142/uialertviews-uiactionsheets-and-keywindow-problems
- 
- Any help would be greatly appreciated.
- */
+@interface UIApplication (window)
 
++ (nullable UIWindow *)currentWindow;
+
+@end
+
+@implementation UIApplication (window)
+
++ (UIWindow *)currentWindow {
```

Comment:
This is a temporary solution where we get around the compiler check on the use of `sharedApplication` property. We need to figure our a long term solution that we can completely avoid `sharedApplication` method in extensions.

However, I have checked out the LTH passcode code. It is not easy to stay away from `sharedApplication`. We would need a fundamental redesign of the LTH passcode view to remove the dependency of `UIApplication`.

Reply:
There is a possible that in future iOS releases the [UIApplication sharedApplication] returns nil for extension targets. So we need to setup a plan to move aways from UIApplication in extensions.